python/cudf/cudf/core/column/column.py

```diff
@@ -994,6 +994,7 @@ def as_string_column(self, dtype, **kwargs):
     def __cuda_array_interface__(self):
         output = {
             "shape": (len(self),),
+            "strides": (self.dtype.itemsize,),
```

Comment:
I don't believe this will work for `CategoricalDtype`?

Reply:
Likely we need to add an `itemsize` property to `CategoricalDtype` to resolve

Reply:
Thanks for catching that. Sure that sounds like a reasonable solution. Where would this go? Also how would we determine the `itemsize` in that case?

Reply:
Good question. The `CategoricalDtype` data type is defined here:

https://github.com/rapidsai/cudf/blob/branch-0.12/python/cudf/cudf/core/dtypes.py#L7

Since the `CategoricalDtype` doesn't know anything about the `codes` of the CategoricalColumn, it actually *wouldn't* make sense to use its `itemsize`. 

 For now, we might have to specialize `__cuda_array_interface__` for `CategoricalColumn` with the `"strides"` set to `self.data.dtype.itemsize`. Does that seem reasonable?

Reply:
Sounds ok to me. Though you are the expert here. 🙂

Should we update `typestr` as well?

Reply:
Yes, actually I'd just do something like:

```
class CategoricalColumn:
    
    @property
    def __cuda_array_interface__(self):
        return self.cat().codes.__cuda_array_interface__
```

Reply:
Ok. Just for my own knowledge what is `codes` here?

Reply:
A CategoricalColumn is just a dictionary encoded column, with the `codes` being the encoded values:

```python
In [1]: import pandas as pd
In [2]: a = pd.Series(['a', 'b', 'a', 'd', 'e'], dtype='category')
In [3]: a.dtype
CategoricalDtype(categories=['a', 'b', 'd', 'e'], ordered=False)
In [4]: a.cat.codes
0    0
1    1
2    0
3    2
4    3
dtype: int8
```

Reply:
Ok, this makes sense.

Reply:
How would we handle this for the other things that are inheriting from `TypedColumnBase`?

Reply:
`__cuda_array_interface__` for strings isn't currently supported, and `NumericalColumn` and `DatetimeColumn` should just work with the `TypedColumnBase` implementation.

Reply:
Right, but how do we handle `itemsize` for `NumericalColumn`  and `DatetimeColumn` cases?

Reply:
Those will be just `self.dtype.itemsize` as you have right now

Reply:
Sorry - maybe I'm misunderstanding?

Reply:
Sorry it's probably me who is misunderstanding. Just trying to wrap my head around how things are structured here. Thanks for bearing with me.

Have pushed the fix you proposed. Please let me know if there is anything else 🙂